src/bin/cli/init.rs

```diff
@@ -205,6 +205,9 @@ pub fn init(args: InitArgs) -> Result<(), Box<dyn Error>> {
             let anchor_version = cargo["dependencies"]["anchor-lang"].clone();
             cargo["dependencies"]["anchor-lang"] = anchor_version.clone();
             cargo["dependencies"]["anchor-spl"] = anchor_version;
+            // TODO move this somewhere else - ideally, Pyth should only be
+            // included as a dependency if it's used in the code
+            cargo["dependencies"]["pyth-sdk-solana"] = Item::Value(Value::String(Formatted::new("0.6.1".to_string())));
```

Comment:
Could we use an optional dependency or feature for this? From https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies

Not sure how you'd do it in code like this, but if we can generate a dependency like this:

```toml
[dependencies]
pyth-sdk-solana = { version = "0.6.1", optional = true }
```

Then there'd be an implicit feature `dep:pyth-sdk-solana` which we could enable/disable at build time. Or we could specifically add a `pyth` feature and tie the dependency to that. Then I'm guessing anchor would have a way to pass features to enable in the `anchor build` command maybe?

Reply:
I was under the impression that unused code wouldn't cause any problems (besides increasing the initial compilation time, which is definitely a bit annoying especially as more things get added).

However, this (https://users.rust-lang.org/t/why-cant-rustc-only-compile-code-that-is-actually-used/64601/6) discussion says that unused code doesn't get compiled to IR and thus doesn't show up in the final binary, unless a library explicitly exports the code with a `pub`. And since the public `seahorse_util` module actually does the Pyth imports, it'll still show up. Whoops.

I think that your suggestion could make managing Cargo.toml a bit easier, but at the cost of the build steps a little less transparent (probably doesn't affect many people though, sometimes I do an `anchor build` for testing). Big problem here is still that we need some system that allows dependencies to be tracked and smartly added/removed as necessary. And if we end up compiling the code without Pyth, then the contents of `seahorse_util` needs to change, adding further complexity.

Reply:
> Big problem here is still that we need some system that allows dependencies to be tracked and smartly added/removed as necessary.

Could we use the import of `seahorse_pyth.py` in Python source as our dependency control? So if that's imported then we activate the pyth feature when we build the program, else we don't. I think we'd be able to detect that now using the code you added to deal with local imports maybe? 

> And if we end up compiling the code without Pyth, then the contents of seahorse_util needs to change, adding further complexity.

I think cargo features should make this reasonably straightforward, it's analogous to how we skip some seahorse code in wasm. But instead of being checked when you compile seahorse, it'd be a feature in the generated code, checked at program compile time

I think it'd look something like:

```rs
// generate/mod.rs
pub mod seahorse_util {
    use super::*;
    use std::{collections::HashMap, fmt::Debug, ops::Deref};
    
    // Re-export for ease of access
    #[cfg(feature="pyth")]
    pub use pyth_sdk_solana::{load_price_feed_from_account_info, PriceFeed};

    ...
```

> I think that your suggestion could make managing Cargo.toml a bit easier, but at the cost of the build steps a little less transparent (probably doesn't affect many people though, sometimes I do an anchor build for testing)

Separately from this PR maybe we could add a `--verbose` or `--debug` flag to make seahorse print the shell commands it's running? Then if we add complexity to them we could easily see what's going on and copy commands for manual testing.

Reply:
> Could we use the import of seahorse_pyth.py in Python source as our dependency control?

Yeah, it would have to work this way. Probably would be another thing similar to how programs are added to instruction contexts by inference. (It's not going to be _that_ hard, the problem is essentially just implementation.)

> I think cargo features should make this reasonably straightforward, it's analogous to how we skip some seahorse code in wasm.

> Separately from this PR maybe we could add a --verbose or --debug flag to make seahorse print the shell commands it's running?

Good call! I think these two ideas combined would make for a good build system. I'll go forward with adding it like this and see how we like it.

Reply:
(which means I might end up adding the --verbose/debug flag this PR lol, it'll be like 5 lines so whatever)

Reply:
Hmm, there actually is a small problem with using the Pyth import as the indicator for adding the Pyth dependency. If someone decides to just `import seahorse` then they can still use `seahorse.pyth` but the compile won't catch the dependency.

Now, I'm pretty sure this would be extremely rare and I've been pretty okay with adding small bugs for the sake of quickly pushing out functionality, but I felt like I should mention this even if we decide to ignore it.

Reply:
Hmm good point! My first thought was could we treat `import seahorse` as enable all features, `import seahorse.pyth` as enable just pyth? But it might be too aggressive to enable new features if someone just did `import seahorse` and then updates (eg to a version after we added something like clockwork with a similar approach). Should we just block `import seahorse` and require the submodules be explicitly used? Not sure how fiddly all this is with the code we have for imports though! 

I think as long as the default for a new project remains `from seahorse.prelude import *` you're right that it should be super rare though!

Reply:
True. I think I'd rather produce that tiny bit of extra bloat than an error for people who do things like `import seahorse` - thoughts?

Definitely don't want to block the functionality though.